lib/jira/client/index.js

```diff
@@ -107,13 +132,31 @@ module.exports = async (id, installationId, jiraHost) => {
         delete: (repositoryId) => instance.delete(`/rest/devinfo/0.10/repository/${repositoryId}`, {
           fields: { _updateSequenceId: Date.now() }
         }),
-        update: (data, options) => instance.post('/rest/devinfo/0.10/bulk', {
-          preventTransitions: (options && options.preventTransitions) || false,
-          repositories: [data],
-          properties: {
-            installationId
+        update: (data, options) => {
+          // Deduplicate issue keys in commits and branches before request
+          if ('commits' in data) data.commits = dedupIssueKeys(data.commits)
+          if ('branches' in data) {
+            data.branches = dedupIssueKeys(data.branches)
+            data.branches.forEach(branch => {
+              if ('lastCommit' in branch) {
+                dedupIssueKeys([branch.lastCommit])
```

Comment:
One other recommendation: sometime function that mutates can be surprising. Is it possible to refactor this code so the dedup and the mutation are separate and more obvious?

```
const dedup = (array) => { [...new Set(array)] }

const dedupRepository = (repositoryObj) => {
  // ...
  if ('commits' in data) {
    data.commits.forEach((commit) => {
      commit.issueKeys = dedup(commit.issueKeys)
    })
  }  
}

// later
dedupRepository(repositoryObj)
```

`dedupRepository` still mutates the object but the mutation is front and centre.